design-toolkit/components/src/components/drawer/drawer.stories.tsx

```diff
@@ -0,0 +1,343 @@
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { Cancel, ChevronLeft, Placeholder } from '@accelint/icons';
+import type { Meta, StoryObj } from '@storybook/react';
+import { useCallback, useState } from 'react';
+import { Button } from '../button';
+import { Icon } from '../icon';
+import { NavigationStack } from '../navigation-stack';
+import { Drawer } from './index';
+import type { DrawerProps } from './types';
+
+const meta: Meta<DrawerProps> = {
+  title: 'Components/Drawer',
+  component: Drawer,
+  parameters: {
+    layout: 'fullscreen',
+  },
+  args: {
+    id: 'left-drawer',
+    placement: 'left',
+    size: 'medium',
+  },
+  argTypes: {
+    size: {
+      control: 'select',
+      options: ['small', 'medium', 'large'],
+    },
+    placement: {
+      control: 'select',
+      options: ['top', 'bottom', 'left', 'right'],
+    },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof Drawer>;
+
+export const WithTabs: Story = {
+  render: ({ id, ...args }) => {
+    return (
+      <div className='h-screen w-full'>
+        <Drawer.Provider>
+          <Drawer.Layout>
+            <Drawer.Main>
+              <div className='h-dvh bg-surface-raised p-l text-default-light'>
+                {longContent}
+              </div>
+            </Drawer.Main>
+            <Drawer id='settings' {...args}>
+              <Drawer.Menu>
+                <Drawer.Menu.Item id='a'>
+                  <Placeholder />
+                </Drawer.Menu.Item>
+                <Drawer.Menu.Item id='b'>
+                  <Placeholder />
+                </Drawer.Menu.Item>
+              </Drawer.Menu>
+              <Drawer.Content>
+                <Drawer.Header>
+                  <Drawer.Title>Title</Drawer.Title>
+                  <Drawer.Trigger for='settings' behavior='close'>
+                    <Button size='small' variant='flat'>
+                      <Icon>
+                        <Cancel />
+                      </Icon>
+                    </Button>
+                  </Drawer.Trigger>
+                </Drawer.Header>
+                <Drawer.Panel id='a'>A Content</Drawer.Panel>
+                <Drawer.Panel id='b'>B Content</Drawer.Panel>
+              </Drawer.Content>
+            </Drawer>
+          </Drawer.Layout>
+        </Drawer.Provider>
+      </div>
+    );
+  },
+};
+
+export const WithLongContent: Story = {
+  render: () => {
```

Comment:
Need to pass args from the Storybook controls into the component.
```suggestion
  render: (args) => {
```